### PR TITLE
Fix requesting IPv4 addresses

### PR DESCRIPTION
**Pull request by farragar**
Stop http_client from hanging when an ipv4 address is passed as the host

### CODE REVIEW

File: lib/http_client.py [hunk 1]
```diff
@@ -214,3 +217,14 @@ def get(url, **kwargs):
 
 def post(url, **kwargs):
 	return request('POST', url, **kwargs)
+
+def is_ipv4_address(address):
+	octets = address.split('.')
+	if len(octets) == 4:
+		try:
+			valid_octets = [x for x in octets if 0 <= int(x) and int(x) <= 255]
+			return len(valid_octets) == 4
```
Comment (marekventur) [inline lib/http_client.py]: you're already doing this check in line 223

**Comment by marekventur** [comment]
rest looks good to me, but I haven't actually tested it on the badge itself

**Comment by Jonty** [comment]
Nice work! Can you squash these commits into one logical commit, just to keep it clean in the history?

**Comment by marekventur** [comment]
@Jonty github has a button for that now :)

**Comment by Jonty** [comment]
@marekventur I prefer the author of the commits to do it so they can write a proper commit message rather than attempting to summarise what they wanted to say!

**Comment by farragar** [comment]
ta @Jonty @marekventur, updated. Changes from the code review (only checking the length once)  _should_ be fine, but haven't actually tested them on the badge yet. Should be able to tomorrow.

**Comment by marekventur** [comment]
looks good to me. If it works on the badge I'm happy to merge

**Comment by farragar** [comment]
Just got around to testing this on the badge again. Works fine 👍

**Comment by marekventur** [comment]
thanks!

**Pull request merged**